netsim/modules/evpn.py

```diff
@@ -16,52 +49,62 @@ def enable_evpn_af(node: Box, topology: Box) -> None:
     if bn.type in bgp_session and 'evpn' in topology.nodes[bn.name].get('module'):
       bn.evpn = True
 
-def vlan_based_service(vlan: Box, vname: str, node: Box, topology: Box) -> None:
+def get_usable_evpn_asn(topology: Box) -> int:
+  asn = data.get_from_box(topology,'evpn.as') or \
+        data.get_from_box(topology,'vrf.as') or \
+        data.get_from_box(topology,'bgp.as')
+
+  if asn and data.is_true_int(asn):
+    return asn
+
+  common.error(
+    f'Cannot get a usable global AS number to use in EVPN route targets',
+    common.IncorrectValue,
+    'evpn')
+  return 0
+
+def vlan_based_service(vlan: Box, vname: str, topology: Box) -> None:
   evpn  = vlan.evpn
-  epath = f'nodes.{node.name}.vlans.{vname}.evpn'
+  epath = f'vlans.{vname}.evpn'
   evpn.evi = evpn.evi or vlan.id                                    # Default EVI value: VLAN ID
+  asn = get_usable_evpn_asn(topology)
   data.must_be_int(
     evpn,'evi',epath,
     module='evpn',
     min_value=1,max_value=65535)                                    # Check EVI data type in range
-  if not 'rd' in evpn:                                              # Default RD value
-    evpn.rd = f'{node.bgp.router_id}:{vlan.id}'
   for rt in ('import','export'):                                    # Default RT value
     if not rt in evpn:                                              # ... BGP ASN:vlan ID
-      evpn[rt] = [ f"{node.bgp['as']}:{vlan.id}" ]
+      evpn[rt] = [ f"{asn}:{vlan.id}" ]
```

Comment:
vlan.id -> evi.id

In case the user manually defines an evi value different from the vlan.id

Reply:
Good catch! Thank you!